### PR TITLE
Implemented dyncmdlist docker container.

### PR DESCRIPTION
**Pull request by JaxkDev**
As requested here's the dynamic command list implementation,
Runs the docker image on every build that is able to produce a phar.
If the plugin is able to load&enable on server startup it will have its commands and aliases gathered and recorded.

### CODE REVIEW

File: src/poggit/ci/builder/DefaultProjectBuilder.php [hunk 1]
```diff
@@ -195,6 +199,65 @@ protected function build(Phar $phar, RepoZipball $zipball, WebhookProjectModel $
...
+        if($pluginName === null){
+            Meta::getLog()->e("Failed to start dyncmdlist, no plugin name found.");
+            return;
+        }
+
+        $id = "dyncmdlist-" . substr(Meta::getRequestId() ?? bin2hex(random_bytes(8)), 0, 4) . "-" . bin2hex(random_bytes(4));
+
+        Meta::getLog()->v("Starting dyncmdlist flow with ID '{$id}'");
+
+        Lang::myShellExec("docker run --name {$id} --cpus=1 --memory=256M -v ${pharPath}:/input/plugin.phar pmmp/dyncmdlist:0.1.1 ./wrapper.sh ${pluginName}", $stdout, $stderr, $exitCode);
```
Comment (SOF3) [inline src/poggit/ci/builder/DefaultProjectBuilder.php]: Is `--rm` missing here?

File: src/poggit/ci/builder/DefaultProjectBuilder.php [hunk 2]
```diff
@@ -195,6 +199,65 @@ protected function build(Phar $phar, RepoZipball $zipball, WebhookProjectModel $
...
+
+        if($exitCode !== 0){
+            Meta::getLog()->e("dyncmdlist failed with code '{$exitCode}', stdout: {$stdout}, stderr: {$stderr}");
+            return;
+        }
+
+        $result = json_decode($stdout, true);
+
+        if($result === null || !isset($result["status"])){
+            Meta::getLog()->e("Failed to parse results from dyncmdlist, stdout: {$stdout}, stderr: {$stderr}");
```
Comment (SOF3) [inline src/poggit/ci/builder/DefaultProjectBuilder.php]: should we check `json_last_error_msg()` here?

File: src/poggit/ci/builder/DefaultProjectBuilder.php [hunk 3]
```diff
@@ -195,6 +199,65 @@ protected function build(Phar $phar, RepoZipball $zipball, WebhookProjectModel $
...
+        }
+
+        if($result["status"] === false){
+            Meta::getLog()->e("dyncmdlist failed with error '{$result["error"]}'");
+            return;
+        }
+
+        if(sizeof($result["commands"]) === 0) {
+            Meta::getLog()->v("dyncmdlist found no commands.");
+            return;
```
Comment (SOF3) [inline src/poggit/ci/builder/DefaultProjectBuilder.php]: are you sure we should return here?

**Review by SOF3 — CHANGES_REQUESTED** [review]

File: src/poggit/ci/builder/DefaultProjectBuilder.php (hunk 1 above)
Comment (JaxkDev) [reply src/poggit/ci/builder/DefaultProjectBuilder.php]: Yes, removed it for debugging earlier.....

File: src/poggit/ci/builder/DefaultProjectBuilder.php (hunk 3 above)
Comment (JaxkDev) [reply src/poggit/ci/builder/DefaultProjectBuilder.php]: Don’t have to but wouldn’t do anything after 🤷‍♂️

File: src/poggit/ci/builder/DefaultProjectBuilder.php (hunk 1 above)
Comment (SOF3) [reply src/poggit/ci/builder/DefaultProjectBuilder.php]: Also just to confirm, is `${pluginName}` shell escape safe?

Comment (JaxkDev) [reply src/poggit/ci/builder/DefaultProjectBuilder.php]: Nope, straight out of plugin.yml iirc

File: src/poggit/ci/builder/DefaultProjectBuilder.php (hunk 2 above)
Comment (JaxkDev) [reply src/poggit/ci/builder/DefaultProjectBuilder.php]: Yeah, had stdout because that's what's being decoded

**Review by SOF3 — APPROVED** [review]

**Pull request merged**